### PR TITLE
MINOR: Replace EasyMock with Mockito in connect:basic-auth-extension

### PR DESCRIPTION
**Pull request by dengziming**
*More detailed description of your change*
Replace EasyMock with Mockito in connect:basic-auth-extension

### CODE REVIEW

**Comment by dengziming** [comment]
ping @ijuma @mageshn to have a look.

File: connect/basic-auth-extension/src/test/java/org/apache/kafka/connect/rest/basic/auth/extension/BasicAuthSecurityRestExtensionTest.java [hunk 1]
```diff
@@ -55,20 +58,21 @@ public void tearDown() {
...
-        EasyMock.expect(configurable.register(EasyMock.capture(jaasFilter))).andReturn(null);
-  
-        ConnectRestExtensionContext context = EasyMock.mock(ConnectRestExtensionContext.class);
-        EasyMock.expect(context.configurable()).andReturn((Configurable) configurable);
-
-        EasyMock.replay(configurable, context);
-  
+        ArgumentCaptor<JaasBasicAuthFilter> jaasFilter = ArgumentCaptor.forClass(JaasBasicAuthFilter.class);
+        Configurable<? extends Configurable<?>> configurable = mock(Configurable.class);
+        when(configurable.register(any())).thenReturn(null);
```
Comment (ijuma) [inline connect/basic-auth-extension/src/test/java/org/apache/kafka/connect/rest/basic/auth/extension/BasicAuthSecurityRestExtensionTest.java]: Why did we not use the captor here?

File: connect/basic-auth-extension/src/test/java/org/apache/kafka/connect/rest/basic/auth/extension/JaasBasicAuthFilterTest.java [hunk 2]
```diff
@@ -53,110 +55,130 @@
     public void testSuccess() throws IOException {
         File credentialFile = setupPropertyLoginFile(true);
         JaasBasicAuthFilter jaasBasicAuthFilter = setupJaasFilter("KafkaConnect", credentialFile.getPath());
-        ContainerRequestContext requestContext = setMock("Basic", "user", "password", false);
+        ContainerRequestContext requestContext = setMock("Basic", "user", "password");
         jaasBasicAuthFilter.filter(requestContext);
```
Comment (ijuma) [inline connect/basic-auth-extension/src/test/java/org/apache/kafka/connect/rest/basic/auth/extension/JaasBasicAuthFilterTest.java:59]: Don't we need any `verify` calls here?

File: connect/basic-auth-extension/src/test/java/org/apache/kafka/connect/rest/basic/auth/extension/BasicAuthSecurityRestExtensionTest.java (hunk 1 above)
Comment (dengziming) [reply connect/basic-auth-extension/src/test/java/org/apache/kafka/connect/rest/basic/auth/extension/BasicAuthSecurityRestExtensionTest.java]: Thank you for your reminders, moving captor from`Mockito.verify` to here is simpler.

File: connect/basic-auth-extension/src/test/java/org/apache/kafka/connect/rest/basic/auth/extension/JaasBasicAuthFilterTest.java (hunk 2 above)
Comment (dengziming) [reply connect/basic-auth-extension/src/test/java/org/apache/kafka/connect/rest/basic/auth/extension/JaasBasicAuthFilterTest.java:59]: Yes, firstly I think it's good enough to verify `jaasBasicAuthFilter.filter(requestContext)` pass without any exception, after some consideration, I think it's better to call `verify` for a double-check.

**Review by ijuma — APPROVED** [review]
LGTM, thanks!

**Pull request merged**